File: mobile/src/main/java/net/mm2d/dmsexplorer/domain/entity/DirectoryEntity.java
/*
 * Copyright (c) 2017 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.dmsexplorer.domain.entity;

import java.util.List;

import androidx.annotation.Nullable;

/**
 * @author <a href="mailto:ryo@mm2d.net">大前良介 (OHMAE Ryosuke)</a>
 */
public interface DirectoryEntity {
    String getParentName();

    boolean isInProgress();

    List<ContentEntity> getEntities();

    void setSelectedEntity(@Nullable ContentEntity entity);

    ContentEntity getSelectedEntity();
}

File: mobile/src/main/java/net/mm2d/dmsexplorer/domain/entity/DirectoryEntity.kt
/*
 * Copyright (c) 2017 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.dmsexplorer.domain.entity

/**
 * @author [大前良介 (OHMAE Ryosuke)](mailto:ryo@mm2d.net)
 */
interface DirectoryEntity {
    val parentName: String
    val isInProgress: Boolean
    val entities: List<ContentEntity>
    var selectedEntity: ContentEntity?
}
